# HG changeset patch
# User Rik <rik@octave.org>
# Date 1380844099 25200
#      Thu Oct 03 16:48:19 2013 -0700
# Node ID 225ec9a0222a9fcbd4c5962dc944ea0553b97491
# Parent  6363302926854e831883c6d0d1c10b60f38128bb
legend.m: Use axis that legend refers to when calculating color for surface patch label.

* scripts/plot/legend.m: Use axis that legend refers to when calculating color
for surface patch label.

diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -875,30 +875,32 @@ function [hleg, hleg_obj, hplot, labels]
                 p1 = patch ("xdata", ([0, linelength, linelength, 0] +
                                       xoffset + xk * xstep) / lpos(3),
                             "ydata", (lpos(4) - yoffset -
                                       [yk-0.3, yk-0.3, yk+0.3, yk+0.3] .* ystep) / lpos(4),
                            "facecolor", facecolor, "edgecolor", edgecolor,
                            "cdata", cdata, "userdata", hplots(k));
                 hobjects(end+1) = p1;
               endif
+              ## FIXME: Probably need listeners, as for line objects
 
             case "surface"
               facecolor = get (hplots(k), "facecolor");
               edgecolor = get (hplots(k), "edgecolor");
-              cdata = sum (caxis ()) / 2;
+              cdata = sum (get (ca(1), "clim")) / 2;
               if (! strcmp (facecolor, "none") || ! strcmp (edgecolor, "none"))
                 p1 = patch ("xdata", ([0, linelength, linelength, 0] +
                                       xoffset + xk * xstep) / lpos(3),
                             "ydata", (lpos(4) - yoffset -
                                       [yk-0.3, yk-0.3, yk+0.3, yk+0.3] .* ystep) / lpos(4),
                            "facecolor", facecolor, "edgecolor", edgecolor,
                            "cdata", cdata, "userdata", hplots(k));
                 hobjects(end+1) = p1;
               endif
+              ## FIXME: Probably need listeners, as for line objects
 
           endswitch
 
           set (texthandle(k), "position",
                               [(txoffset + xk * xstep) / lpos(3), ...
                                (lpos(4) - yoffset - yk * ystep) / lpos(4)]);
           if (strcmp (orientation, "vertical"))
             yk++;
